service/src/main/java/com/codeforcommunity/dataaccess/AuthDatabaseOperations.java

```diff
@@ -90,4 +93,47 @@ public boolean isOnBlackList(String signature) {
             Tables.BLACKLISTED_REFRESHES
                 .where(Tables.BLACKLISTED_REFRESHES.REFRESH_HASH.eq(signature)));
     }
+
+    public void createSecretKey(int userId, String token) throws AuthException {
+        if (!db.fetchExists(Tables.NOTE_USER.where(NOTE_USER.ID.eq(userId)))) {
+            throw new AuthException("User does not exist.");
+        }
+
+        VerificationKeysRecord keysRecord = db.newRecord(Tables.VERIFICATION_KEYS);
+        keysRecord.setId(token);
+        keysRecord.setUserId(userId);
+        keysRecord.store();
+    }
+
+    private boolean isTokenDateValid(VerificationKeysRecord tokenResult) {
+        Timestamp cutoffDate = Timestamp.from(Instant.now().minusSeconds(CUTOFF_TIME));
+        if (tokenResult.getCreated().before(cutoffDate)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public void validateSecretKey(String secretKey) throws AuthException {
+        VerificationKeysRecord veriKey = db.selectFrom(Tables.VERIFICATION_KEYS)
+            .where(Tables.VERIFICATION_KEYS.ID.eq(secretKey)
+                .and(Tables.VERIFICATION_KEYS.USED.eq((short)0)))
+            .fetchOneInto(VerificationKeysRecord.class);
+
+        if (veriKey == null) {
+            throw new AuthException("Token is invalid.");
+        }
+
+        if (!isTokenDateValid(veriKey)) {
+            throw new AuthException("Token has expired.");
+        }
+
+        veriKey.setUsed((short)1);
+        veriKey.store();
+        NoteUserRecord noteUser = db.selectFrom(Tables.NOTE_USER)
+            .where(Tables.NOTE_USER.ID.eq(veriKey.getUserId()))
+            .fetchOneInto(NoteUserRecord.class);
+        noteUser.setVerified((short)1);
```

Comment:
I see what you were talking about with having to update one field. Looking at it now I think you were right about just writing it as an update statement rather than using records. Sorry for the wrong advice